pytests/func/test_django.py

```diff
@@ -0,0 +1,190 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in the project root
+# for license information.
+
+from __future__ import print_function, with_statement, absolute_import
+
+import os.path
+
+from ..helpers.pattern import ANY
+from ..helpers.timeline import Event
+from .testfiles.testroots import get_test_root
+from ..helpers.webhelper import get_url_from_str, get_web_content
+
+DJANGO1_ROOT = get_test_root('django1')
+DJANGO1_MANAGE = os.path.join(DJANGO1_ROOT, 'app.py')
+DJANGO1_TEMPLATE = os.path.join(DJANGO1_ROOT, 'templates', 'hello.html')
+
+def _wait_for_django_link(debug_session):
+    o = None
+    while True:
+        if o:
+            o = debug_session.wait_for_next(o >> Event('output'))
+        else:
+            o = debug_session.wait_for_next(Event('output'))
+        for e in debug_session.all_occurrences_of(Event('output')):
+            link = get_url_from_str(o.body['output'])
+            if link is not None:
+                return link
+
+
+def _django_no_multiproc_common(debug_session):
+    debug_session.multiprocess = False
+    debug_session.cli_args = ['runserver', '--noreload', '--nothreading']
+
+    debug_session.ignore_unobserved += [
+        # The queue module can spawn helper background threads, depending on Python version
```

Comment:
The comment is probably wrong here. Might just say that Django starts a lot of helper threads, and the test is too high-level to bother with them.